tests/func/test_file.py

```diff
@@ -43,3 +46,27 @@ def test_resolve_file_no_exist(cloud_test_catalog):
     assert resolved_non_existent.size == 0
     assert resolved_non_existent.etag == ""
     assert resolved_non_existent.last_modified == TIME_ZERO
+
+
+def test_upload(cloud_test_catalog):
+    ctc = cloud_test_catalog
+
+    src_uri = ctc.src_uri
+    filename = "image_1.jpg"
+    source = f"{src_uri}/upload-test-images"
+    catalog = ctc.catalog
+
+    image = Image.new(mode="RGB", size=(64, 64))
+    img_bytes = io.BytesIO()
+    image.save(img_bytes, format="JPEG")
+    img_bytes = img_bytes.getvalue()
+    assert img_bytes
```

Comment:
This _setup_ is not necessary for the `upload` test. 

```suggestion
    img_bytes = b"bytes"
```

Reply:
The reason that I used the elaborate setup was for the test to demonstrate the expected usage of the API.